### ISSUE TITLE
Still receiving messages after calling unsubscribe() on subscription

### ISSUE THREAD

**Issue by cascornelissen** [stomp-js/stompjs#370]
Hi there! 👋🏼 

We seem to be running into the same issue as [documented here on StackOverflow](https://stackoverflow.com/questions/46228123/ng2-stomp-client-unsubscribe-not-working/46229026) and since that hasn't received an answer I'm hoping someone here can help. The following workaround mentioned in the comments doesn't work in our case, we really need the subscriptions on a single connection.

> I ended up simply opening multiple connections and creating a single subscription on each, allowing me to stop the data flow of individual subscriptions by disconnecting the specific connection. There must be a better way, but I couldn't find it.

We're using `@stomp/stompjs@6.1.0` and `sockjs-client@1.5.1`.

---

We do the following:

1. Create a client and activate it
2. Once the `onConnect` callback is triggered we subscribe to a bunch of different topics, ids for the subscriptions get assigned internally.
3. We can see the `SUBSCRIBE` message being sent out and start receiving incoming messages with content
    ```
    ["SUBSCRIBE\nid:sub-0\ndestination:/topic/live-mode/redacted\n\n\u0000"]
    ```
    ```
    a["MESSAGE\ndestination:/topic/live-mode/redacted\ncontent-type:application/json\nsubscription:sub-0\nmessage-id:mfqvbriu-8\ncontent-length:52\n\n{\"timestamp\":\"2021-08-09T09:05:59.800Z\",\"value\":0.0}\u0000"]
    ```
4. We click a button to unsubscribe (this triggers `subscription.unsubscribe()` on multiple subscriptions) and see the `UNSUBSCRIBE` messages being sent out.
    ```
    ["UNSUBSCRIBE\nid:sub-0\n\n\u0000"]
    ```

So far, everything seems to be going according to plan but after the `UNSUBSCRIBE` event, with the correct id, is sent out the back-end keeps returning messages. I checked this with them and they're saying that event never arrives. The back-end is using Spring and its internal Websocket/STOMP support.

Any ideas on how to further debug this would be great.

---

We're currently investigating further and noticed the following:

* We can't seem to reproduce the issue with [`stomp-websocket`](https://www.npmjs.com/package/stomp-websocket) while it seems to be sending out the exact same messages.
* Copy/pasting the `SUBSCRIBE` and `UNSUBSCRIBE` messages into Postman also seems to work.

There's also a [question on StackOverflow](https://stackoverflow.com/questions/68653392/spring-websocket-doesnt-listen-to-unsubscribe-event-with-stomp-stompjs) from the back-end side with more information.

**Comment by kum-deepak** [comment]
I am out of my wits here. Some information:

- Up to the v4, this library was an extended version of the original stomp-websocket.
- v5 was rewritten bottom up, including the parser and other infrastructure.
- The Web STOMP implementation at Spring seems to be developed quite a lot in conjunction with the original stomp-websocket. I am aware of one feature, beyond STOMP standards, is the handling of large messages - only supported by Spring and this library.
- So, it may be that code at Spring is tested only with stomp-websocket and not against the STOMP specs.
- From v5, when in conflict, this library follows the STOMP specs.

In summary, I have no clue what is going wrong.

The lowest level one can check is the WebSocket tab in Chrome. If no issues are detected there, the Spring side checks may be needed,

One hunch I have - the stomp-websocket, always uses WebSocket Text frames, the current version of this library may use Text or Binary WebSocket frames. As Java treats binary and text data differently, the Spring behaviour for text and binary frames may not use the same code paths.

Nonetheless please attach your STOMP config.

**Comment by cascornelissen** [comment]
Thanks so much for the quick reply and for looking into this, also for the context on `stomp-websocket`, much appreciated! We spent a couple more hours on this problem and eventually figured out that the issue is (likely) as follows:

1. A `SUBSCRIBE` message is sent out from the client
2. The back-end registers the subscription but also immediately polls for data from another service
3. The `UNSUBSCRIBE` message is sent out from the client
    1. If this message arrives on the back-end after the additional request from step 2 has completed everything works as expected
    2. If this message arrives on the back-end before the additional request from step 2 has completed it can't do anything because the subscription apparently is not known internally yet.
        This basically means that for the back-end the `UNSUBSCRIBE` comes in first and then the `SUBSCRIBE` comes in later, as the `SUBSCRIBE` messages are handled in series.

We've disabled the additional request from step 2 and it works without any problems so we're currently trying to re-implement the logic that fetches additional data asynchronously. Disclaimer: I have no experience with Java/Spring so this description might be slightly off but the general problem should be clear.

I'll close the issue with 🤞🏼 that our back-end developers are able to resolve this issue on their side know that we know what the root cause of this issue is.
